base/replutil.jl

```diff
@@ -515,6 +515,7 @@ function show_method_candidates(io::IO, ex::MethodError, kwargs::Vector=Any[])
                     # If the methods args is longer than input then the method
                     # arguments is printed as not a match
                     for (k, sigtype) in enumerate(sig[length(t_i)+1:end])
+                        sigtype = unwrap_unionall(sigtype)
```

Comment:
I think this should only be done in the `isvarargtype` case. Otherwise we won't correctly show argument types that are UnionAlls.

This code also doesn't seem to handle arguments written as `::Vararg{T,N}`, but that's a separate issue.